### PR TITLE
Create inkstitch.rb

### PR DESCRIPTION
**Pull request by grischard**
[Inkstitch](https://inkstitch.org/) is an Inkscape extension for machine embroidery design. Dear maintainer, if you want homebrew hats, you need this!

I am parsing the json from the github API for the livecheck. I see no other cask doing this, is it a bad idea?

Also, while this isn't my first contribution, it's my first new cask. Comments and suggestions for improvement very welcome.

---

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [X] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [X] `brew audit --cask <cask>` is error-free.
- [X] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [X] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by grischard** [comment]
I didn't know about `github_latest` which isn't documented. Thanks!

**Comment by miccal** [comment]
Thank you @grischard, but I do not think this will work via Cask:
```
/usr/bin/sudo -E -- /usr/bin/env LOGNAME=runner USER=runner USERNAME=runner /usr/sbin/installer -pkg /usr/local/Caskroom/inkstitch/2.1.2/inkstitch-v2.1.2-osx.pkg -target /
==> Installing artifact of class Cask::Artifact::Pkg
==> Running installer for inkstitch; your password may be necessary.
Package installers may write to any location; options such as `--appdir` are ignored.
installer: Package name is inkstitch-v2.1.2-osx
installer: Installing at base path /
Error: Failure while executing; `/usr/bin/sudo -E -- /usr/bin/env LOGNAME=runner USER=runner USERNAME=runner /usr/sbin/installer -pkg /usr/local/Caskroom/inkstitch/2.1.2/inkstitch-v2.1.2-osx.pkg -target /` exited with 1. Here's the output:
installer: The install failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “inkstitch-v2.1.2-osx.pkg”.)
```

I am guessing this is a result of this:

Screen Shot 2022-02-03 at 12 45 36

I will leave this open in case other maintainers of @Homebrew/cask have any ideas.

**Comment by grischard** [comment]
The postinstall script is quite simple:

`mv /tmp/inkstitch/ $HOME/Library/Application\ Support/org.inkscape.Inkscape/config/inkscape/extensions/`

Would a zap that removes that satisfy the homebrew cask guidelines?

**Comment by miccal** [comment]
No that will not work -- the problem here is that the `.pkg` will not even install in the first place on our CI machines, hence the `uninstall` is not even called.

**Comment by miccal** [comment]
Possibly relevant: https://github.com/inkstitch/inkstitch/issues/1531

**Comment by miccal** [comment]
I think the problem is that Inkscape needs to be actually run so that the folder `~/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions/` actually exists.

**Comment by carlocab** [comment]
[This](https://github.com/inkstitch/inkstitch/blob/be93c858a4b7bdd771521502c3c1195a01b32fbf/bin/build-distribution-archives#L42-L50) probably needs to be fixed:

```sh
        /usr/bin/pkgbuild --root dist/inkstitch.app \
                        -s "${INSTALLER_IDENT}" \
                        --component-plist installer_scripts/inkstitch.plist \
                        --ownership recommended \
                        --identifier org.inkstitch.installer \
                        --version ${VERSION} \
                        --scripts installer_scripts/scripts \
                        --install-location ${temp_path}inkstitch.app \
                        artifacts/inkstitch-${VERSION}-${OS}.pkg
```
where
```
    temp_path="/tmp/inkstitch/"
```

File: Casks/inkstitch.rb [hunk 1]
```diff
@@ -30,7 +30,9 @@
   end
 
   preflight do
-    system_command "/bin/mkdir", args: ["-p", "#{ENV["HOME"]}/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions"]
+    system_command "/bin/mkdir",
+                   args: ["-p", "#{ENV["HOME"]}/Library/Application Support \
+                   /org.inkscape.Inkscape/config/inkscape/extensions"]
```
Comment (carlocab) [inline Casks/inkstitch.rb:35]:
```suggestion
                   args: [
                     "-p",
                     "#{ENV["HOME"]}/Library/Application Support/org.inkscape.Inkscape/config/inkscape/extensions",
                   ]
```
Seems easier to read. Unless this isn't standard in cask.

Comment (carlocab) [reply Casks/inkstitch.rb:35]: Ah, wait, this probably still violates the line length limit.

Comment (carlocab) [inline Casks/inkstitch.rb:35]:
```suggestion
                   args: ["-p", "#{ENV["HOME"]}/Library/Application Support\
                                 /org.inkscape.Inkscape/config/inkscape/extensions"]
```
You don't want that last space, I think. Might be nicer to align the string continuation too.

Comment (miccal) [reply Casks/inkstitch.rb]: Thanks @carlocab :)

**Comment by miccal** [comment]
Ok, seems to be working now.

@grischard while we wait for other maintainers to have there say regarding the `preflight` block I have had to add here to get this to work, could you please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap), thanks.

**Comment by grischard** [comment]
As far as I can tell, only `/Library/Application\ Support/inkstitch` exists, and is empty here. The other content would be user documents, templates, etc. where deletion would raise eyebrows.

**Comment by miccal** [comment]
Let's go ahead and merge this, and see how it goes.

**Pull request merged**